chore: add markdownlint, markdown link check workflow and with updated markdown file

xuan-cao-swi opened the pull request:
## Description
This PR attempts to address https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/547 (remove gitter) and https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/981.

1. A markdown link check, similar to the one used by [opentelemetry.io](https://github.com/open-telemetry/opentelemetry.io/blob/main/.markdown-link-check.json), has been added. This check was run as a sample and identified a few dead links, which have been rectified in this PR.
2. The markdown lint has been configured to ignore the `CHANGELOG.md` file (e.g., `!**/CHANGELOG.md`). This is because `CHANGELOG.md` is believed to be auto-generated based on PR titles. Any issues with this file could potentially cause the entire workflow to fail.
3. The Gitter communication platform has been removed and replaced with a Slack channel. This change is aimed at improving the communication process. 

Closes #547
Closes #981

kaylareopelle commented on .github/workflows/ci-markdown-link.yml:
```diff
@@ -0,0 +1,17 @@
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "MarkDown Link Check"
```
```suggestion
      - name: "Markdown Link Check"
```

kaylareopelle commented on .github/workflows/ci-markdownlint.yml:
```diff
@@ -0,0 +1,20 @@
+      - uses: actions/checkout@v4
+
+      # equivalent cli: markdownlint-cli2  "**/*.md" "#**/CHANGELOG.md"  --config .markdownlint.json
+      - name: "MarkDown Lint Check"
```
```suggestion
      - name: "Markdown Lint Check"
```

kaylareopelle commented on .instrumentation_generator/templates/Readme.md.tt:
```diff
@@ -38,7 +38,7 @@ Example usage can be seen in the `./example/trace_demonstration.rb` file [here](
 The `opentelemetry-instrumentation-<%= instrumentation_name %>` gem source is [on github][repo-github], along with related gems including `opentelemetry-api` and `opentelemetry-sdk`.
 
-The OpenTelemetry Ruby gems are maintained by the OpenTelemetry-Ruby special interest group (SIG). You can get involved by joining us in [GitHub Discussions][discussions-url] or attending our weekly meeting. See the [meeting calendar][community-meetings] for dates and times. For more information on this and other language SIGs, see the OpenTelemetry [community page][ruby-sig].
+The OpenTelemetry Ruby gems are maintained by the OpenTelemetry-Ruby special interest group (SIG). You can get involved by joining us on our [GitHub Discussions][discussions-url], [Slack Channel][slack-channel] or attending our weekly meeting. See the [meeting calendar][community-meetings] for dates and times. For more information on this and other language SIGs, see the OpenTelemetry [community  ...
```
nit: I know this was here before, but since we're updating this paragraph anyway, `OpenTelemetry-Ruby` looks weird to me. I prefer `OpenTelemetry Ruby`. 🤷 

It's just a style thing I'm seeing, but it doesn't have to change.

xuan-cao-swi replied:
Thanks, I have changed to OpenTelemetry Ruby

kaylareopelle reviewed (approved):
Thanks, @xuan-cao-swi! Few little things I saw, but this is great as-is. Happy we're getting these issues squared away. 

Could you update your PR description to link the issues so that they close when this PR is merged? 

The update could look something like:
```
Closes #547
Closes #981
```

Here's a doc on how the linking feature works: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

xuan-cao-swi commented:
Looks like [https://github.com/open-telemetry/community/](https://github.com/open-telemetry/community/blob/master/CONTRIBUTING.md) have a series of change :confused:, and I couldn't find the replacement link (e.g. https://github.com/open-telemetry/community/blob/master/CONTRIBUTING.md).

It seems like they just removed some of the link like https://github.com/open-telemetry/community/blob/master/community-membership.md#maintainer, and replace it directly with ruby repo [contributing](https://github.com/open-telemetry/opentelemetry-ruby#contributing)

kaylareopelle commented:
> Looks like [https://github.com/open-telemetry/community/](https://github.com/open-telemetry/community/blob/master/CONTRIBUTING.md) have a series of change 😕, and I couldn't find the replacement link (e.g. https://github.com/open-telemetry/community/blob/master/CONTRIBUTING.md).
> 
> It seems like they just removed some of the link like https://github.com/open-telemetry/community/blob/master/community-membership.md#maintainer, and replace it directly with ruby repo [contributing](https://github.com/open-telemetry/opentelemetry-ruby#contributing)
> 
> Link has been fixed.

Strange! I used that community-membership link often. I'll open an issue on the repo to see if we can figure out what's replaced those two files. I think you have a good workaround for now.

kaylareopelle commented:
@xuan-cao-swi  - The membership doc lives here now: https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md 

Could you add it back to the files?

xuan-cao-swi commented:
> @xuan-cao-swi - The membership doc lives here now: https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md
> 
> Could you add it back to the files?

Yes, I changed the link to https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md. Sorry for the confusion, I mean I fixed the link in our README for `Link has been fixed.`
